File: 62. Unique Paths/dp.java
class Solution {
    public int uniquePaths(int m, int n) {
        
        int[][] dp = new int[m][n];
        dp[0][0] = 1;
        
        for(int y = 0; y < m; y++){
            for(int x = 0; x < n; x++){
                if(y == 0 && x == 0) continue;
                
                dp[y][x] = 0;
                
                if(y > 0) dp[y][x] += dp[y-1][x];
                if(x > 0) dp[y][x] += dp[y][x-1];
            }
        }
        
        return dp[m-1][n-1];
    }
}

File: 0518. Coin Change 2/dp.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        dp = [[0 for j in range(amount+1)] for i in range(len(coins))];
        
        for i in range(len(coins)):
            for j in range(amount + 1):
                if j == 0:
                    dp[i][j] = 1;
                else:
                    coin = coins[i];
                    top = dp[i-1][j] if i > 0 else 0
                    left = dp[i][j - coin] if j - coin >= 0 else 0;
                    dp[i][j] = top + left;
       
        return dp[len(coins) - 1][amount]